src/plugins/controls/public/control_group/editor/control_group_editor.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiFlyoutHeader,
+  EuiButtonGroup,
+  EuiFlyoutBody,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiFlyoutFooter,
+  EuiButton,
+  EuiFormRow,
+  EuiButtonEmpty,
+  EuiSpacer,
+  EuiCheckbox,
+} from '@elastic/eui';
+
+import { ControlGroupStrings } from '../control_group_strings';
+import { ControlStyle, ControlWidth } from '../../types';
+import { CONTROL_LAYOUT_OPTIONS, CONTROL_WIDTH_OPTIONS } from './editor_constants';
+
+interface EditControlGroupProps {
+  width: ControlWidth;
+  controlStyle: ControlStyle;
+  setAllWidths: boolean;
+  updateControlStyle: (controlStyle: ControlStyle) => void;
+  updateWidth: (newWidth: ControlWidth) => void;
+  updateAllControlWidths: (newWidth: ControlWidth) => void;
+  onCancel: () => void;
+  onClose: () => void;
+}
+
+export const ControlGroupEditor = ({
```

Comment:
Following the logic of the editor for individual controls (`control_editor.tsx`), I pulled out the flyout logic so that it can manage its own state internally - this not only helped make things more consistent, it also fixed a bug I introduced when I tried to use the control group editor as an API in `getEditControlGroupButton` (from `control_group_container.tsx`)